File: static/time.js
let app = new Vue({
    el: '#app',
    data: {
        hour: 0,
        minute: 0,
        second: 0,
        date: '',
        offset: 0,
        interval: null
    },
    template: /*html*/`
        <div>
            <h1>{{ date }}</h1>
            <h2>{{ hour }}:{{ minute }}:{{ second }}</h2>
        </div>
    `,
    created: function() {
        this.offset = (new Date()).getTimezoneOffset() * -1;
        this.sync();
        this.interval = setInterval(this.tick, 864);

        document.addEventListener('visibilitychange', () => {
            if (document.visibilityState === 'visible') {
                this.sync();
            }
        });
    },
    methods: {
        sync: function() {
            fetch('/now?offset=' + this.offset)
            .then(response => response.json())
            .then(data => {
                const rep_time = data._embedded['repcal:time'];
                const rep_date = data._embedded['repcal:date'];
                this.hour = rep_time.attributes.hour;
                this.minute = rep_time.attributes.minute;
                this.second = rep_time.attributes.second;
                this.date = rep_date.texts.default;
            });
        },
        tick: function() {
            this.second++;

            if (this.second === 100) {
                this.second = 0;
                this.minute++;

                if (this.minute === 100) {
                    this.minute = 0;
                    this.hour++;

                    if (this.hour === 10) {
                        this.hour = 0;
                        this.sync();
                    }
                }
            }
        }
    }
});

File: resources/Time.py
from . import Resource
from datetime import time
from repcal import DecimalTime


class Time(Resource):
    def __init__(self, t: time) -> None:
        self.time = t
        self.decimal = DecimalTime.from_standard_time(t)

    def type(self) -> str: return 'time'

    def params(self) -> dict:
        return {
            'hour': self.time.hour,
            'minute': self.time.minute,
            'second': self.time.second
        }

    def to_dict(self) -> dict:
        return {
            "texts": {
                "default": str(self.decimal),
                "decimal": self.decimal.decimal
            },
            "attributes": {
                "hour": self.decimal.hour,
                "minute": self.decimal.minute,
                "second": self.decimal.second
            }
        }
